### PR TITLE
docs(argo-cd): Document various ingress setups

### PR DESCRIPTION
**Pull request by pdrastil**
Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Review by mbevc1 — APPROVED** [review]

File: charts/argo-cd/README.md [hunk 1]
```diff
@@ -64,7 +64,170 @@ applicationSet:
   replicas: 2
 ```
 
-### Synchronizing Changes from Original Repository
+## Ingress configuration
+
+Please refer to [Operator Manual](https://argo-cd.readthedocs.io/en/stable/operator-manual/ingress/#ingress-configurationh) for details as samples below
+corespond to respetive sections.
```
Comment (tico24) [inline charts/argo-cd/README.md:70]: >respetive

typo

File: charts/argo-cd/README.md [hunk 2]
```diff
@@ -64,7 +64,170 @@ applicationSet:
...
+
+### SSL-Passthrough
+
+The `tls: true` option will expect that `argocd-server-tls` secret exists as Argo CD server loads TLS certificates from this place.
```
Comment (tico24) [inline charts/argo-cd/README.md:74]: should be "will expect that the `argocd`..."

File: charts/argo-cd/README.md [hunk 3]
```diff
@@ -64,7 +64,170 @@ applicationSet:
...
+    enabled: true
+    hostname: argocd.example.com
+    ingressClassName: nginx
+    annotations:
+      nginx.ingress.kubernetes.io/force-ssl-redirect: "true"
+      nginx.ingress.kubernetes.io/backend-protocol: "HTTP"
+    extraTls:
+      - hosts:
+        - argocd.example.com
+        # Secret might be optional based on used ingress controller
```
Comment (tico24) [inline charts/argo-cd/README.md:110]: better to say "based on the ingress controller used".

That said, this example is very nginx-specific anyway.

File: charts/argo-cd/README.md [hunk 4]
```diff
@@ -64,7 +64,170 @@ applicationSet:
...
+        secretName: wildcard-tls
+```
+
+> **Note:**
+> It's also possible to use `tls: true` without `extraTls` if you don't plan to use wildcard certificate.
```
Comment (tico24) [inline charts/argo-cd/README.md:115]: should be "if you don't plan on using the wildcard certificate."

File: charts/argo-cd/README.md [hunk 5]
```diff
@@ -64,7 +64,170 @@ applicationSet:
...
+
+### Multiple ingress resources for gRPC protocol support
+
+Use `ingressGrpc` section if your ingress controller supports only single protocol per Ingress resource (i.e.: Contour).
```
Comment (tico24) [inline charts/argo-cd/README.md:119]: 1. should be "supports only a single"

2. should 'Ingress' have an uppercase I here? it's not consistent.

File: charts/argo-cd/README.md [hunk 6]
```diff
@@ -64,7 +64,170 @@ applicationSet:
...
+      nginx.ingress.kubernetes.io/backend-protocol: "HTTPS"
+    tls: true
+    extraHosts:
+      - name: argocd-alias.example.com
+        path: /
+```
+
+### AWS Application Load Balancer
+
+Refer to Operator Manual for [AWS Application Load Balancer mode](https://argo-cd.readthedocs.io/en/stable/operator-manual/ingress/#aws-application-load-balancers-albs-and-classic-elb-http-mode).
```
Comment (tico24) [inline charts/argo-cd/README.md:165]: should be "Refer to the Operator Manual"

**Review by yu-croco — APPROVED** [review]

File: charts/argo-cd/README.md [hunk 7]
```diff
@@ -64,7 +64,170 @@ applicationSet:
...
+Provided example assumes you are using TLS off-loading via AWS ACM service.
```
Comment (tico24) [inline charts/argo-cd/README.md:166]: should be "The provided example assumes..."

File: charts/argo-cd/README.md [hunk 8]
```diff
@@ -64,7 +64,170 @@ applicationSet:
...
+
+> **Note:**
+> Using `controller: aws` creates additional service for gRPC traffic and it's no longer needed to use `ingressGrpc` configuration section.
```
Comment (tico24) [inline charts/argo-cd/README.md:169]: should be "no longer need to use the `ingress`...."

File: charts/argo-cd/README.md [hunk 9]
```diff
@@ -64,7 +64,170 @@ applicationSet:
...
+      alb.ingress.kubernetes.io/listen-ports: '[{"HTTPS":80}, {"HTTPS":443}]'
+      alb.ingress.kubernetes.io/ssl-redirect" '443'
+    aws:
+      serviceType: ClusterIP # <- Used with target-type: ip
+      backendProtocolVersion: GRPC
+```
+
+### GKE Application Load Balancer
+
+Implementation will populate `ingressClassName`, `networking.gke.io/managed-certificates` and `networking.gke.io/v1beta1.FrontendConfig` annotations
```
Comment (tico24) [inline charts/argo-cd/README.md:195]: should be "The implementation will..."

File: charts/argo-cd/README.md.gotmpl [hunk 10]
```diff
@@ -63,7 +63,171 @@ applicationSet:
   replicas: 2
 ```
 
-### Synchronizing Changes from Original Repository
+## Ingress configuration
+
+Please refer to [Operator Manual](https://argo-cd.readthedocs.io/en/stable/operator-manual/ingress/#ingress-configurationh) for details as samples below
```
Comment (tico24) [inline charts/argo-cd/README.md.gotmpl:68]: should be "Please refer to the..." and "as the samples below"

File: charts/argo-cd/README.md.gotmpl [hunk 11]
```diff
@@ -63,7 +63,171 @@ applicationSet:
...
+corespond to respetive sections.
```
Comment (tico24) [inline charts/argo-cd/README.md.gotmpl:69]: > respetive

typo

File: charts/argo-cd/README.md [hunk 12]
```diff
@@ -64,7 +64,170 @@ applicationSet:
   replicas: 2
 ```
 
-### Synchronizing Changes from Original Repository
+## Ingress configuration
+
+Please refer to [Operator Manual](https://argo-cd.readthedocs.io/en/stable/operator-manual/ingress/#ingress-configurationh) for details as samples below
```
Comment (tico24) [inline charts/argo-cd/README.md:69]: should be "Please refer to the..." and "as the samples below"

File: charts/argo-cd/README.md.gotmpl [hunk 13]
```diff
@@ -63,7 +63,171 @@ applicationSet:
...
+
+### SSL-Passthrough
+
+The `tls: true` option will expect that `argocd-server-tls` secret exists as Argo CD server loads TLS certificates from this place.
```
Comment (tico24) [inline charts/argo-cd/README.md.gotmpl:73]: I've just remembered that all this will be the same as above so I won't re-review it!

**Review by tico24 — CHANGES_REQUESTED** [review]
some nits

File: charts/argo-cd/README.md (hunk 1 above)
Comment (tico24) [reply charts/argo-cd/README.md:70]: should also probably be "to their respective sections.

Comment (pdrastil) [reply charts/argo-cd/README.md]: Fixed

File: charts/argo-cd/README.md (hunk 2 above)
Comment (pdrastil) [reply charts/argo-cd/README.md]: Fixed

File: charts/argo-cd/README.md (hunk 3 above)
Comment (pdrastil) [reply charts/argo-cd/README.md]: Fixed

File: charts/argo-cd/README.md (hunk 5 above)
Comment (pdrastil) [reply charts/argo-cd/README.md]: 1) Fixed.
2) I've used Ingress with capital I for K8s resource and ingress controller for deployment

File: charts/argo-cd/README.md (hunk 9 above)
Comment (pdrastil) [reply charts/argo-cd/README.md]: Fixed

File: charts/argo-cd/README.md (hunk 4 above)
Comment (tico24) [reply charts/argo-cd/README.md]: or maybe "a wildcard certificate" now that I think about it. (sorry)

**Review by tico24 — APPROVED** [review]

**Comment by pdrastil** [comment]
@tico24 Thanks for corrections 👍

**Review by jmeridth — APPROVED** [review]

**Pull request merged**